### PR TITLE
Pixel Likelihood Calculations

### PR DESCRIPTION
**Pull request by ParsonsRD**
This is an implementation of the pixel likelihood calculation as described in:
https://arxiv.org/abs/0907.2610

Both the gaussian approximation and full likelihood calculation are implemented. As well as the calculation of the likelihood expectation value.

### CODE REVIEW

File: ctapipe/image/pixel_likelihood.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+class PixelLikelihoodError(RuntimeError):
+    pass
+
+
+def poisson_likelihood_gaussian(image, prediction, spe_width, ped):
+    """
+    Calculate likelihood of prediction given the measured signal, gaussian approx from
+    de Naurois et al 2009
+    Parameters
```
Comment (kosack) [inline ctapipe/image/pixel_likelihood.py]: Probably need a blank line before the section title

File: ctapipe/image/pixel_likelihood.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    ----------
+    image: ndarray
+        Pixel amplitudes from image
+    prediction: ndarray
+        Predicted pixel amplitudes from model
+    spe_width: ndarray
+        width of single p.e. distributio
+    ped: ndarray
+        width of pedestal
+    Returns
```
Comment (kosack) [inline ctapipe/image/pixel_likelihood.py]: insert blank lin

File: ctapipe/image/pixel_likelihood.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        Predicted pixel amplitudes from model
+    spe_width: ndarray
+        width of single p.e. distribution
+    ped: ndarray
+        width of pedestal
+    width_fac: float
+        Factor to determine range of summation on integral
+    dtype: datatype
+        Data type of output array
+    Returns
```
Comment (kosack) [inline ctapipe/image/pixel_likelihood.py]: insert blank line

File: ctapipe/image/pixel_likelihood.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
+# Class for calculation of likelihood of a pixel expectation, given the pixel amplitude
+# and typically the level of noise in the pixel
+import numpy as np
+import math
+from scipy.misc import factorial
+from scipy.integrate import quad
+
+
```
Comment (kosack) [inline ctapipe/image/pixel_likelihood.py]: need to add an `__all__` array here to say which functions you want to export to the `ctapipe.image` namespace (right now you export everything, including the scipy and math libraries, which causes some API documentation issues)

**Comment by ParsonsRD** [comment]
OK that should have got it. I also added the new class to the automatic api generation in the documentation.

**Pull request merged**